### DISCUSSION TITLE
Whats the Virtualenv and user for manual update for an installation set up by the installer script

### DISCUSSION THREAD

**Discussion by nicosqc** [modoboa/modoboa#2713] [Installer]
Hello,

I set up my modoboa instance with the installer script, gave no parameters and didnt tweak the config file, so the instance that was set up did not have Letsencrypt enabled by default. I later on added a systemd timer and service for this myself as well as changing the transport for the internal mails to LMTP as described in #2564. Because of these manual changes I would like to also do a manual update, but I already am struggling with the first step:

> sudo -u <modoboa_user> -i
> source <virtuenv_path>/bin/activate

I checked that the modoboa user is ``modoboa`` but I don't know where to find the virtualenv. My modoboa instance is set up in ``/srv`` and is currently v2.0.1

Also: how can I get a list of all my installed plugins?

**Comment by nicosqc** [comment]
I found 4 files called "activate" on my system, I assume it's the one in ``/srv/modoboa/env/`` but I would just like to double check:

```find / -name "activate" 2> /dev/null
/srv/modoboa/env/bin/activate
/srv/radicale/env/bin/activate
/srv/automx/env/bin/activate
/usr/lib/python3.7/venv/scripts/common/activate

**Comment by tonioo** [comment]
The one you're looking for is /srv/modoboa/env/bin/activate

**Reply by nicosqc** [reply]
Thanks! Could you tell me how I can check which extensions I have installed? I am planning to upgrade the server but I cannot afford a lot of downtime as I have no secondary server set up and the one I want to upgrade is used in production.

**Reply by nicosqc** [reply]
Think I found them in the ``settings.py``:

```
# A dedicated place to register Modoboa applications
# Do not delete it.
# Do not change the order.
MODOBOA_APPS = (
    'modoboa',
    'modoboa.core',
    'modoboa.lib',
    'modoboa.admin',
    'modoboa.transport',
    'modoboa.relaydomains',
    'modoboa.limits',
    'modoboa.parameters',
    'modoboa.dnstools',
    'modoboa.policyd',
    'modoboa.maillog',
    # Modoboa extensions here.
    'modoboa_amavis',
    'modoboa_pdfcredentials',
    'modoboa_postfix_autoreply',
    'modoboa_sievefilters',
    'modoboa_webmail',
    'modoboa_contacts',
    'modoboa_radicale',

)
```

**Reply by nicosqc** [reply]
Do I need to take the server offline for the upgrade or is it safe to attempt while everything is running?

**Reply by nicosqc** [reply]
I have created a backup of all the server files with 
```
sudo -i
mkdir /fullback
sudo rsync -aAXv / --exclude={"/dev/*","/proc/*","/sys/*","/tmp/*","/run/*","/mnt/*","/media/*","/lost+found","/fullback"} /fullback
mkdir /fullback/postgres-backup/
chown postgres:postgres /fullback/postgres-backup
```
and then dumped all databases with ``sudo -i -u postgres pg_dumpall > /fullback/postgres-backup/postgres-dumpall.sql``
Is this sufficient to restore the system from a critical error while upgrading or are there any components that store data in memory or could be in an inconsistent state on the file system?

**Reply by tonioo** [reply]
You can run the upgrade while everything is running but it's also fine to stop services before. Your backup looks ok but if you installed Modoboa using the installer, you can use the new backup feature
